modules/47_regipol/regiCarbonPrice/datainput.gms

```diff
@@ -22,32 +22,35 @@ $IFTHEN.emiMkt not "%cm_emiMktTarget%" == "off"
 *** initialize emiMkt Target parameters
   p47_targetConverged(ttot2,ext_regi) = 0;
 
-*** initialize carbon taxes before start year 
-if ( (cm_startyear gt 2005),
-  Execute_Loadpoint 'input_ref' p47_taxCO2eq_ref = pm_taxCO2eq;
-  Execute_Loadpoint 'input_ref' p47_taxemiMkt_init = pm_taxemiMkt;
-
-*** copying taxCO2eq value to emiMkt tax parameter for fixed years that contain no pm_taxemiMkt value
-  p47_taxemiMkt_init(ttot,regi,emiMkt)$((p47_taxCO2eq_ref(ttot,regi)) and (ttot.val le cm_startyear) and (NOT(p47_taxemiMkt_init(ttot,regi,emiMkt)))) = p47_taxCO2eq_ref(ttot,regi);
-
-*** Initializing European ETS historical and reference prices
-  loop(regi$regi_groupExt("EUR_regi",regi),
-    p47_taxemiMkt_init("2005",regi,"ETS")$(cm_startyear le 2005) = 0;
-    p47_taxemiMkt_init("2010",regi,"ETS")$(cm_startyear le 2010)  = 15*sm_DptCO2_2_TDpGtC;
-    p47_taxemiMkt_init("2015",regi,"ETS")$(cm_startyear le 2015)  = 8*sm_DptCO2_2_TDpGtC;
-***  p47_taxemiMkt_init("2020",regi,"ETS")$(cm_startyear le 2020)  = 41.28*sm_DptCO2_2_TDpGtC; !! 2018 =~ 16.5€/tCO2, 2019 =~ 25€/tCO2, 2020 =~ 25€/tCO2, 2021 =~ 53.65€/tCO2, 2022 =~ 80€/tCO2 -> average 2020 = 40€/tCO2 -> 40*1.032 $/tCO2 = 41.28 $/t CO2
-    p47_taxemiMkt_init("2020",regi,"ETS")$(cm_startyear le 2020)  = 30*sm_DptCO2_2_TDpGtC;
-
-*** Initializing European ESR historical and reference prices
-    p47_taxemiMkt_init("2020",regi,"ES")$(cm_startyear le 2020)  = 30*sm_DptCO2_2_TDpGtC;
-    p47_taxemiMkt_init("2020",regi,"other")$(cm_startyear le 2020)  = 30*sm_DptCO2_2_TDpGtC;
-  );
+*** initialize carbon taxes
+p47_taxemiMkt_init(t,regi,emiMkt) = 0;
+
+Execute_Loadpoint 'input_ref' p47_taxCO2eq_ref = pm_taxCO2eq;
+Execute_Loadpoint 'input_ref' p47_taxemiMkt_init = pm_taxemiMkt;
+
+*** copying taxCO2eq value to emiMkt tax parameter for years and regions that contain no pm_taxemiMkt value
+p47_taxemiMkt_init(t,regi,emiMkt)$(p47_taxCO2eq_ref(t,regi) and (NOT(p47_taxemiMkt_init(t,regi,emiMkt)))) = p47_taxCO2eq_ref(t,regi);
 
-*** intialize price trajectory after 2020 based on historical year prices for non policy scenarios 
-  if ( (cm_startyear le 2020),
-    p47_taxemiMkt_init(t,regi,emiMkt)$(t.val gt 2020)  = p47_taxemiMkt_init("2020",regi,emiMkt) + (cm_postTargetIncrease*sm_DptCO2_2_TDpGtC)*(t.val-2020);
+*** Overwrite historical prices for Europe if they are uncontrolled
```

Comment:
could you briefly explain where the "if they are uncontrolled" part is dealt with in the following code?

Reply:
The `pm_taxemiMkt` is only initialized for historical years if the historical years are free in a `cm_emiMktTarget` run. Otherwise the values contained in the reference scenario gdx are used (these are usually the `pm_taxCO2eqHist` values).

Reply:
https://github.com/remindmodel/remind/pull/1128/commits/f7eff6e6309a8a1d6161c73ceca7eb39d71a9d18